tripal/src/Form/TripalEntityForm.php

```diff
@@ -57,4 +58,30 @@ public function save(array $form, FormStateInterface $form_state) {
     $form_state->setRedirect('entity.tripal_entity.canonical', ['tripal_entity' => $entity->id()]);
   }
 
+  /**
+   *
+   * {@inheritDoc}
+   * @see \Drupal\Core\Entity\EntityForm::actions()
+   */
+  protected function actions(array $form, FormStateInterface $form_state) {
+    $actions = parent::actions($form, $form_state);
+    $route_info = $this->entity->toUrl('unpublish-form');
```

Comment:
This line is what is generating the error - apparently at this point `$this->id()` is NULL.

Reply:
Thanks @dsenalik for hitting on this problem!  When I overwrote the `actions()` function I failed to include a check if the entity was new (similar to the parent function that adds the "edit" and "delete" actions).  I've updated the code and tested a brand new installation and it's working now.  Please feel free to try again.